src/ImageSharp/Formats/Jpeg/JpegThrowHelper.cs

```diff
@@ -8,6 +8,13 @@ namespace SixLabors.ImageSharp.Formats.Jpeg
 {
     internal static class JpegThrowHelper
     {
+        /// <summary>
+        /// Cold path optimization for throwing <see cref="NotSupportedException"/>'s.
+        /// </summary>
+        /// <param name="errorMessage">The error message for the exception.</param>
+        [MethodImpl(InliningOptions.ColdPath)]
+        public static void ThrowNotSupportedException(string errorMessage) => throw new NotSupportedException(errorMessage);
```

Comment:
Do we throw `ImageFormatException` or `NotSupportedException` in other cases we don't support decoding a valid input?

Ideally, we need to make sure to document all exceptions we throw on `Image.Load` overloads, although I'm less strict on this since I know that the BCL is also full of undocumented exceptions :)

Reply:
we throw `ImageFormatException` in cases when the image data is not as expected, e.g. not as the specification suggest.
Those cases here the jpeg files are valid, so i think `NotSupportedException` is the correct Exception to throw.

Reply:
+1 for `NotSupportedException` because corrupted image and not supported image are very different scenarios :)

Reply:
Great to see the docs fixed in `Image.FromFile.cs` and `Image.FromBytes.cs` ! 👍

I would also change the docs in `Image.FromStream`:

```C#
/// <exception cref="NotSupportedException">The stream is not readable. -or- The image format is not supported.</exception>
```